test/Vault.test.js

```diff
@@ -0,0 +1,117 @@
+const { expect } = require('chai')
+const { ethers } = require('@nomiclabs/buidler')
+
+const BN = ethers.BigNumber
+
+describe('Vault', function () {
+    let vault
+    let erc20Mock
+    let owner, user
+    let ownerAddr, userAddr
+    const amount = BN.from(100).mul(BN.from(10).pow(18))
+
+    beforeEach(async function () {
+        const [creator, ownerSigner, userSigner] = await ethers.getSigners()
+        owner = ownerSigner
+        ownerAddr = await owner.getAddress()
+
+        user = userSigner
+        userAddr = await user.getAddress()
+
+        const Vault = await ethers.getContractFactory('Vault', creator)
+
+        vault = await Vault.deploy(await ownerSigner.getAddress())
+        await vault.deployed()
+
+        const ERC20Mock = await ethers.getContractFactory('ERC20Mock')
+        erc20Mock = await ERC20Mock.deploy()
+        await erc20Mock.deployed()
+    })
+
+    it('Can deploy successfully', async function () {
+        expect(vault.address).to.not.equal(0)
+    })
+
+    it('Transfers ownership to address provided', async function () {
+        expect(await vault.owner()).to.equal(ownerAddr)
+    })
+
+    describe('Stake', function () {
```

Comment:
@kwix I want this contract to be more generic. Thus, it doesn't have the staking mechanism or any notion of that.

Tokens are deposited and withdrawn.

My suggestion is to rename "stake" to "deposit".